pkg/gotenberg/pdfengine.go

```diff
@@ -64,6 +70,12 @@ type PdfEngine interface {
 	// Convert transforms a given PDF to the specified formats defined in
 	// PdfFormats. If no format, it does nothing.
 	Convert(ctx context.Context, logger *zap.Logger, formats PdfFormats, inputPath, outputPath string) error
+
+	// ReadMetadata reads the exif metadata of the given PDF files and load into passed metadata object.
+	ReadMetadata(ctx context.Context, logger *zap.Logger, paths []string, metadata []FileMetadata) error
```

Comment:
The ExifTool [read](https://github.com/barasher/go-exiftool/blob/master/exiftool.go#L143-L144)/[write](https://github.com/barasher/go-exiftool/blob/master/exiftool.go#L214) metadata function can do the operation on multiple files in one go.

We also have a use case where users can convert multiple files simultaneously. Shouldn't we process them once instead of calling them one by one for metadata overwrite?

Reply:
Got it.

🤔 that being said, `gotenberg.PdfEngine` is not only about `exiftool`; one could hypothetically provide an implementation for another tool.

Regarding the alternative with a single file, you could handle in each route adding/overriding metadata all files using for instance:

https://github.com/gotenberg/gotenberg/blob/b8926350b9ea4a7cc624bde5a97350c9128a677f/pkg/modules/chromium/events.go#L212. 

Performance wise (i.e., operation speed), I'm pretty sure it would work better. And keep the actual implementation cleaner with less structs.

---

On another subject, some of the comments are about "exit metadata", but, afaik, all metadata are concerned?

Reply:
Hi, thanks for your reply.

I trust your judgment to keep it a single file operation for metadata. I am keeping the process to write metadata to multiple files sequentially just like PDF conversion happens.

I am observing some errors in my tests with multiple files executing simultaneously. It could be an issue with [go-exiftool](https://github.com/barasher/go-exiftool) and I haven't looked into the code.


Regarding comments mentioning only "exif metadata". Yeah, it covers others as well and I am correcting that as well.

Thanks again.